packages/react-reconciler/src/ReactFiberRoot.js

```diff
@@ -106,7 +125,24 @@ export function createFiberRoot(
     expirationTime: NoWork,
     firstBatch: null,
     nextScheduledRoot: null,
+
+    nextGlobalRoot: null,
+    previousGlobalRoot: lastGlobalRoot,
+    setContext: (setRootContext.bind(null, uninitializedFiber): any),
```

Comment:
If we pass the GlobalRoot as an argument to setContext we don't have to bind this function and create a new one for each root. We can share one function per renderer. We can get the fiber from arg.current.

Reply:
Come to think of it, we already do pass it to `this`. So you won't need to bind this function.